test/components/structures/auth/Login-test.tsx

```diff
@@ -253,4 +254,69 @@ describe("Login", function () {
         const ssoButtons = container.querySelectorAll(".mx_SSOButton");
         expect(ssoButtons.length).toBe(idpsWithIcons.length + 1);
     });
+
+    it("should display an error when homeserver doesn't offer any supported login flows", async () => {
+        mockClient.loginFlows.mockResolvedValue({
+            flows: [
+                {
+                    type: "just something weird",
+                },
+            ],
+        });
+
+        getComponent();
+        await waitForElementToBeRemoved(() => screen.queryAllByLabelText("Loading…"));
+
+        expect(
+            screen.getByText("This homeserver doesn't offer any login flows which are supported by this client."),
+        ).toBeInTheDocument();
+    });
+
+    it("should display a connection error when getting login flows fails", async () => {
+        mockClient.loginFlows.mockRejectedValue("oups");
+
+        getComponent();
+        await waitForElementToBeRemoved(() => screen.queryAllByLabelText("Loading…"));
+
+        expect(
+            screen.getByText("There was a problem communicating with the homeserver, please try again later."),
+        ).toBeInTheDocument();
+    });
+
+    it("should display an error when homeserver fails liveliness check", async () => {
+        fetchMock.resetBehavior();
+        fetchMock.get("https://matrix.org/_matrix/client/versions", {
+            status: 400,
+        });
+        getComponent();
+        await waitForElementToBeRemoved(() => screen.queryAllByLabelText("Loading…"));
+
+        // error displayed
+        expect(screen.getByText("Your test-brand is misconfigured")).toBeInTheDocument();
+    });
+
+    it("should reset liveliness error when server config changes", async () => {
+        fetchMock.resetBehavior();
```

Comment:
Do we need these `.resetBehaviours` if they're already in the `beforeEach` hook?

Not familiar with using the package the mocked fetch comes from, so there could be some quirk I'm unaware of here.

Reply:
I found them necessary to reset clear the `/versions` mock that is set in the `beforeEach` hook